vent/api/actions.py

```diff
@@ -34,339 +34,352 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
+                container.remove(force=True)
+                self.logger.info("cleaned "+str(container_name))
+            except Exception as e:
+                self.logger.warning("failed to clean "+str(container_name)+" because: "+str(e))
+        return status
+
+    def build(self,
              repo=None,
              name=None,
              groups=None,
              enabled="yes",
              branch="master",
              version="HEAD"):
-       """ Build a set of tools that match the parameters given """
-       args = locals()
-       options = ['image_name', 'path']
-       status = (True, None)
-       sections, template = self.plugin.constraint_options(args, options)
-       for section in sections:
-           self.logger.info("Building "+str(section)+" ...")
-           template = self.plugin.builder(template, sections[section]['path'],
+        """ Build a set of tools that match the parameters given """
+        args = locals()
+        options = ['image_name', 'path']
+        status = (True, None)
+        sections, template = self.plugin.constraint_options(args, options)
+        for section in sections:
+            self.logger.info("Building "+str(section)+" ...")
+            template = self.plugin.builder(template, sections[section]['path'],
                                         sections[section]['image_name'],
                                         section, build=True, branch=branch,
                                         version=version)
-       template.write_config()
-       return status
-
-   @staticmethod
-   def backup():
-       return
-
-   @staticmethod
-   def restore():
-       return
-
-   @staticmethod
-   def configure():
-       # tools, core, etc.
-       return
-
-   @staticmethod
-   def system_info():
-       return
-
-   @staticmethod
-   def system_conf():
-       return
-
-   @staticmethod
-   def system_commands():
-       # restart, shutdown, upgrade, etc.
-       return
-
-   @staticmethod
-   def logs():
-       return
-
-   @staticmethod
-   def help():
-       return
-
-   def inventory(self, choices=None):
-       # repos, core, tools, images, built, running, etc.
-       # plugins that have been added, built, etc.
-       items = {}
-
-       try:
-           for choice in choices:
-               # !! TODO
-               items[choice] = ()
-       except Exception as e: # pragma: no cover
-           pass
-
-       return items
+        template.write_config()
+        return status
+
+    @staticmethod
+    def backup():
+        return
+
+    @staticmethod
+    def restore():
+        return
+
+    @staticmethod
+    def configure():
+        # tools, core, etc.
+        return
+
+    @staticmethod
+    def system_info():
+        return
+
+    @staticmethod
+    def system_conf():
+        return
+
+    @staticmethod
+    def system_commands():
+        # restart, shutdown, upgrade, etc.
+        return
+
+    def logs(self, grep_list={}):
```

Comment:
per the quantifiedcode issue, make `grep_list=None` instead of an empty dict.  it has an explanation of why this change is a better practice.